src/algebra/group_ring_action.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+
+Group action on rings.
+-/
+
+import group_theory.group_action data.equiv.ring
+
+universes u v
+
+variables (α : Type u) (β : Type v)
+
+section prio
+set_option default_priority 100 -- see Note [default priority]
+
+/-- Typeclass for multiplicative actions by monoids on monoids. -/
+class mul_mul_action [monoid α] [monoid β] extends mul_action α β :=
+(smul_one : ∀ (g : α), (g • 1 : β) = 1)
+(smul_mul : ∀ (g : α) (x y : β), g • (x * y) = (g • x) * (g • y))
+
+/-- Typeclass for multiplicative actions by monoids on additive monoids. -/
+class mul_add_action [monoid α] [add_monoid β] extends mul_action α β :=
+(smul_zero : ∀ (g : α), (g • 0 : β) = 0)
+(smul_add : ∀ (g : α) (x y : β), g • (x + y) = (g • x) + (g • y))
+
+attribute [to_additive mul_add_action] mul_mul_action
+
+set_option old_structure_cmd true
+/-- Typeclass for multiplicative actions by monoids on semirings. -/
+class mul_semiring_action [monoid α] [semiring β] extends mul_mul_action α β, mul_add_action α β.
+end prio
+
+export mul_mul_action (smul_one smul_mul)
+export mul_add_action (smul_zero smul_add)
+
+/-- Each element of the monoid defines a monoid homomorphism. -/
+def mul_mul_action.to_mul_hom [monoid α] [monoid β] [mul_mul_action α β] (x : α) : β →* β :=
```

Comment:
I think that we should introduce `monoid.End M := M →* M` with `* = comp`, and have `α →* monoid.End β` here, though it will require explicit `α` and `β` for `coe_fn` in Lean 3.

Reply:
Done.